pkg/cmd/roachtest/test_runner.go

```diff
@@ -877,6 +877,10 @@ func (r *testRunner) runWorker(
 
 				c.goCoverDir = t.GoCoverArtifactsDir()
 
+				if c.Spec().WorkloadNode {
+					c.Put(ctx, t.DeprecatedWorkload(), "./workload", c.WorkloadNode())
```

Comment:
For consideration: we're making every test pay the price of uploading the workload binary for a handful of tests that actually need them (mostly (only?) schemachange IIRC). Maybe we could add a `UseDeperecatedWorkload` option to enable this conditionally.

Ignorable suggestion that can be done in future work.

Reply:
Hmm, but almost all tests that use workload currently call  some version of `c.Put(ctx, t.DeprecatedWorkload(), "./workload", c.Node(nodes+1))`. That was my main intent of automatically uploading it so we could remove that boilerplate.

Or do you mean that we could/should refactor most `workload` calls to be `cockroach workload` instead? In which case I agree, happy to do that in a followup.

Reply:
>While a lot smaller than the cockroach binary, the workload binary is still 100MB+.

What if we stop uploading the cockroach binary onto the workload node, so it's a net negative for stuff we need to upload? Or would we rather just move away from `deprecatedWorkload` completely. If so I can work on switching everything to `cockroach workload` while I'm switching them to use `spec.WorkloadNode` (currently WIP as a followup PR).

Reply:
I suppose the first approach would still punish the "good tests" 😓 

I'm okay making that change, gonna revert the mandatory `c.Put` of workload and fix it in a followup PR unless anyone has objections.

Reply:
Yeah I think we should work towards removing `DeprecatedWorkload`. _Only_ uploading the `workload` binary would go against that.